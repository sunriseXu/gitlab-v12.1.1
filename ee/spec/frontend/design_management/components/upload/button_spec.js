import { shallowMount } from '@vue/test-utils';
import UploadButton from 'ee/design_management/components/upload/button.vue';

describe('Design management upload button component', () => {
  let vm;

  function createComponent(isSaving = false, isInverted = false) {
    vm = shallowMount(UploadButton, {
      propsData: {
        isSaving,
        isInverted,
      },
    });
  }

  afterEach(() => {
    vm.destroy();
  });

  it('renders upload design button', () => {
    createComponent();

    expect(vm.element).toMatchSnapshot();
  });

  it('renders inverted upload design button', () => {
    createComponent(false, true);

    expect(vm.element).toMatchSnapshot();
  });

  it('renders loading icon', () => {
    createComponent(true);

    expect(vm.element).toMatchSnapshot();
  });

  describe('onFileUploadChange', () => {
    it('emits upload event', () => {
      createComponent();

      jest.spyOn(vm.find({ ref: 'fileUpload' }).element, 'files', 'get').mockReturnValue('test');

      vm.vm.onFileUploadChange('test');

      expect(vm.emitted().upload[0]).toEqual(['test']);
    });
  });

  describe('openFileUpload', () => {
    it('triggers click on input', () => {
      createComponent();

      const clickSpy = jest.spyOn(vm.find({ ref: 'fileUpload' }).element, 'click');

      vm.vm.openFileUpload();

      expect(clickSpy).toHaveBeenCalled();
    });
  });
});
