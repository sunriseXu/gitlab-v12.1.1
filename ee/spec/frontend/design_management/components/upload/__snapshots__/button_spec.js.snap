// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management upload button component renders inverted upload design button 1`] = `
<div>
  <glbutton-stub
    class="btn-inverted"
    variant="primary"
  >
    
    Upload designs
    
    <!---->
  </glbutton-stub>
   
  <input
    accept="image/*"
    class="hide"
    multiple="multiple"
    name="design_file"
    type="file"
  />
</div>
`;

exports[`Design management upload button component renders loading icon 1`] = `
<div>
  <glbutton-stub
    class=""
    disabled="true"
    variant="primary"
  >
    
    Upload designs
    
    <glloadingicon-stub
      class="ml-1"
      color="orange"
      inline="true"
      label="Loading"
      size="sm"
    />
  </glbutton-stub>
   
  <input
    accept="image/*"
    class="hide"
    multiple="multiple"
    name="design_file"
    type="file"
  />
</div>
`;

exports[`Design management upload button component renders upload design button 1`] = `
<div>
  <glbutton-stub
    class=""
    variant="primary"
  >
    
    Upload designs
    
    <!---->
  </glbutton-stub>
   
  <input
    accept="image/*"
    class="hide"
    multiple="multiple"
    name="design_file"
    type="file"
  />
</div>
`;
